python/dgl/contrib/dis_kvstore.py

```diff
@@ -449,3 +721,116 @@ def get_id(self):
             KVClient ID
         """
         return self._client_id
+
+
+    def get_addr(self):
+        """Get client IP address
+
+        Return
+        ------
+        str
+            IP address
+        """
+        return self._addr
+
+
+    def _get_local_addr(self):
+        """Get local available IP and port
+
+        Return
+        ------
+        str
+            IP address, e.g., '192.168.8.12:50051'
+        """
+        s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+        try:
+            # doesn't even have to be reachable
+            s.connect(('10.255.255.255', 1))
+            IP = s.getsockname()[0]
+        except:
+            IP = '127.0.0.1'
+        finally:
+            s.close()
+        
+        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        s.bind(("",0))
+        s.listen(1)
+        port = s.getsockname()[1]
+        s.close()
+
+        return IP + ':' + str(port)
+
+
+    def _takeId(self, elem):
+        """Used by sort
+        """
+        return elem.rank
+
+
+    def _push_handler(self, name, ID, data, target):
+        """Default handler for local PUSH message. 
+
+        On default, _push_handler perform SET operation for the tensor.
+
+        Parameters
+        ----------
+        name : str
+            data name
+        ID : tensor (mx.ndarray or torch.tensor)
+            a vector storing the ID list.
+        data : tensor (mx.ndarray or torch.tensor)
+            a tensor with the same row size of id
+        target : tensor (mx.ndarray or torch.tensor)
+            the target tensor
+        """
+        target[name][ID] = data
```

Comment:
we definitely don't want people to have user-defined logic on both clients and servers. we need to fix it later.